scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -145,13 +145,45 @@ class FossId(
             .build()
 
         service = retrofit.create(FossIdRestService::class.java)
+
+        version = runBlocking {
+            parseVersion().orEmpty()
+        }
     }
 
     override fun filterOptionsForResult(options: ScannerOptions) =
         options.mapValues { (k, v) ->
             v.takeUnless { k in secretKeys }.orEmpty()
         }
 
+    /**
+     * Extract the version version from the login page.
+     * Example: &nbsp;&nbsp;&nbsp;cli.  3.1.16 (build 5634934d, RELEASE)
+     */
+    private suspend fun parseVersion(): String? {
+        // TODO: replace with an API call when FossID provides a function (starting at version 21.2).
+        val regex = Regex("^.*&nbsp;(cli. *[0-9\\. ]+\\(build[\\w, ]+\\)).*$")
+
+        val response = service.getLoginPage()
+
+        response.use {
```

Comment:
I'm wondering whether this outer `use` on the response body is even needed, as all that closing the response does is to close the body, which probably is what closing the buffered stream below already does. Maybe you can verify by debugging?

Reply:
I was just wondering the same :)
Yes `useLines` closes the `BufferedReader` even if early returned.

Reply:
And does the `BufferedReader` also close the underlying response body?

Reply:
Yes, I tested it. The `Source` deep within the `ResponseBody` is also closed.